Adds Dependencies for Fedora and Arch (#94)
cemu-project/Cemu: BUILD.md

For ubuntu and most derivatives:

`sudo apt install -y libgtk-3-dev libsecret-1-dev libgcrypt20-dev libsystemd-dev freeglut3-dev clang-12 nasm ninja-build`

For Arch and most derivatives:

`sudo pacman -S --needed gtk3 libsecret libgcrypt systemd-libs freeglut nasm git zlib vcpkg zip unzip`

For Fedora and most derivatives:

`sudo dnf in -y gtk3-devel libsecret-devel libgcrypt-devel systemd-devel freeglut-devel nasm cmake vcpkg perl-core zlib-devel cubeb-devel`

1) Run `git clone --recursive https://github.com/cemu-project/Cemu`
2) `cd Cemu`
3) `mkdir build && cd build`
4) `cmake .. -DCMAKE_BUILD_TYPE=release -DCMAKE_C_COMPILER=/usr/bin/clang-12 -DCMAKE_CXX_COMPILER=/usr/bin/clang++-12 -G Ninja`

> 5) `ninja`

Reviewer: Pardon my interjection, but using `1.` (instead of `1)`) is the correct way to create numbered lists in Markdown.

Reviewer: Just left a comment as I was poking around with the code/docs to see if there's anything I could provide myself, apologies in advance if a whole review comes across as a bit aggressive.

Reviewer 2: That package install list for Fedora won't install ```llvm``` or ```clang```
Also why can't you change ```dnf in``` to ```dnf install``` ? 
I get that the former will work, but using shorthand for official documentation doesn't seem like the best approach.

Author: i am leaving it according to how it was by exzap for now.

Author: > That package install list for Fedora won't install `llvm` or `clang`
> Also why can't you change `dnf in` to `dnf install` ?
> I get that the former will work, but using shorthand for official documentation doesn't seem like the best approach.

sudo dnf in is just a better approach fedora users are used to it plus people tend to copy and paste. 

Clang came pre installed on 36/37 for me and i didn’t need llvm its why it didn’t get if you can show it does then i can add it.

Reviewer 2: from ```man dnf```
```
       Available commands:

       • install
```
The command is called ```install``` and it's simply not a better approach to use a short-hand name for that in official documentation.  Also clang is not installed by default in any build I've seen. And it's definitely  not included in the 'Fedora Workstation'' environmental group. Yes you need to install ```llvm``` as well if you're using ```clang```. I can show you....just need a few minutes....

Author: > from `man dnf`
> 
> ```
>        Available commands:
> 
>        • install
> ```
> 
> The command is called `install` and it's simply not a better approach to use a short-hand name for that in official documentation. Also `clang` is not installed by default in any build I've seen. And it's definitely not included in the 'Fedora Workstation' environmental group. Also, you shouldn't assume what packages someone might already have installed as you can build this from any number of environments including a Fedora podman container. You should also add `vulkan-headers` and `zip` (will install `unzip` as a dependency) to the install list. That's odd, I just compiled Cemu without needing llvm. A recent PR must have fixed that. Ok, it looks like we don't need that any longer.

Never ever had to install llvm it worked without installing not sure what your doing but it’s not right and in is fine I won’t be changing it

Reviewer 2: Do you actually read what I wrote?!  
No offense, but I don't think you're the right person for this job. 
I specifically stated ```it looks like we don't need that any longer```
As in you do not need to install ```llvm```
I don't know how you could have possibly misinterpret that. Just do whatever you're going to do. I'll correct your shortcomings later.

Reviewer 3: > Pardon my interjection, but using `1.` (instead of `1)`) is the correct way to create numbered lists in Markdown.

1) works too

Reviewer: @Tachi107 On **GitHub's** *custom* flavor of Markdown. It's not a part of the Markdown specification.

For ubuntu and most derivatives:

`sudo apt install -y libgtk-3-dev libsecret-1-dev libgcrypt20-dev libsystemd-dev freeglut3-dev clang-12 nasm ninja-build`

For Arch and most derivatives:

`sudo pacman -S cmake git base-devel ninja nasm linux-headers zip spirv-headers`

For Fedora and most derivatives:

`sudo dnf in cmake ninja-build vcpkg llvm kernel-headers gtk3-devel libsecret-devel libgcrypt-devel systemd-devel freeglut-devel nasm perl-core zlib-devel cubeb-devel`

1) Run `git clone --recursive https://github.com/cemu-project/Cemu`
2) `cd Cemu`
3) `mkdir build && cd build`
4) `cmake .. -DCMAKE_BUILD_TYPE=release -DCMAKE_C_COMPILER=/usr/bin/clang -DCMAKE_CXX_COMPILER=/usr/bin/clang++ -G Ninja`

4.5) To build for gcc `cmake .. -DCMAKE_BUILD_TYPE=release -DCMAKE_C_COMPILER=/usr/bin/gcc -DCMAKE_CXX_COMPILER=/usr/bin/g++ -G Ninja -DCMAKE_MAKE_PROGRAM=/usr/bin/ninja`

6) `ninja`

> Due to issue with clang and how ubuntu handles it's filesystem and libs some of

Reviewer: There's a certain amount of typos here (such as `its`), might be best to cite what you mean using `[text](https://example.com)` and this as a reference: https://wiki.ubuntu.com/MultiarchSpec

Author: i will fix thanks

> Due to issue with clang and how ubuntu handles it's filesystem and libs some of
> You might not have audio and an option is to use pulseaudio instead of pipewire 
> But this is a very bad idea and could cause more issues with stability. 

Reviewer 3: I don't understand what you are referring to when talking about Ubuntu's handling of "filesystem and libs", nor how this paragraph can be useful to users having the issue. It might be more helpful to add `libpulse-dev` to the list of needed dependencies of Debian and Ubuntu, as mentioned in #40.

```suggestion
Due to issue with clang and how Ubuntu handles its filesystem and libs some of
you might not have audio. An option is to use PulseAudio instead of PipeWire,
but this could cause more issues with stability. 
```

@n0toose

> There's a certain amount of typos here (such as its), might be best to cite what you mean using `[text](https://example.com)` and this as a reference: https://wiki.ubuntu.com/MultiarchSpec

Why is Multiarch related to this paragraph?

Author: i tried libpulse-dev in fedora and arch and audio still didn't work. if you want to add it to the list make it a PR to my repo i will accept i just don't touch ubuntu. it has caused me enough problems in life.

For ubuntu and most derivatives:

> `sudo apt install -y libgtk-3-dev libsecret-1-dev libgcrypt20-dev libsystemd-dev freeglut3-dev clang-12 nasm ninja-build`

Reviewer 3: I think that `cmake` is required too :)

Author: It’s not

Reviewer 3: How are you going to run `cmake` on step 4 if `cmake` is not installed?

Author: Because it comes with ubuntu … already and this pr isn’t about ubuntu title says its all.

Reviewer 3: > Because it comes with ubuntu … already

No, it doesn't, `cmake` is an optional package.

 > this pr isn’t about ubuntu title says its all. 

Since you're already modifying the file you could make life simpler also for Ubuntu users, no? :)

For Arch and most derivatives:

`sudo pacman -S cmake git base-devel ninja nasm linux-headers zip spirv-headers`

Vcpkg provides spirv-headers but for some it's needed in arch it seems. 

Added zip incase the user runs KDE but with gnome zip is included.

> vcpkg is not needed to be installed and provided by cemu when git cloning.

Reviewer 3: `incase` -> `in case`

The three statements could be modified to be a bit clearer.

For Fedora and most derivatives:

`sudo dnf in cmake ninja-build vcpkg llvm kernel-headers gtk3-devel libsecret-devel libgcrypt-devel systemd-devel freeglut-devel nasm perl-core zlib-devel cubeb-devel`

> 1) Run `git clone --recursive https://github.com/cemu-project/Cemu`

Reviewer 3: `https://github.com/cemu-project/Cemu.git`

2) `cd Cemu`
3) `mkdir build && cd build`

> 4) `cmake .. -DCMAKE_BUILD_TYPE=release -DCMAKE_C_COMPILER=/usr/bin/clang -DCMAKE_CXX_COMPILER=/usr/bin/clang++ -G Ninja`

Reviewer 3: `cmake -B build` is a bit more concise than `mkdir build && cd build && cmake ..`

If you change to what I suggested, 6th step should be either `ninja -C build` or `cmake --build build`

Author: 6 ) it just ninja no ninja -c or anything weird 

and the rest is already correct and works nothing left to change

Reviewer 3: Yes, it works already. I just pointed out that `cmake -B build` is a bit more concise :)

Author: Again keep on topic this pr is only about arch and fedora deps the cmake stuff is how it should be and won’t be changed i only added gcc to go with the new master pr that was merged.

> 4.5) To build for gcc `cmake .. -DCMAKE_BUILD_TYPE=release -DCMAKE_C_COMPILER=/usr/bin/gcc -DCMAKE_CXX_COMPILER=/usr/bin/g++ -G Ninja -DCMAKE_MAKE_PROGRAM=/usr/bin/ninja`

Reviewer 3: I don't think that `-DCMAKE_MAKE_PROGRAM` is required

Author: It’s required

Reviewer 3: From the [documentation](https://cmake.org/cmake/help/latest/variable/CMAKE_MAKE_PROGRAM.html):

> Tool that can launch the native build system.
> ...
> The [`Ninja`](https://cmake.org/cmake/help/latest/generator/Ninja.html#generator:Ninja) generator sets this to `ninja`.

The `Ninja` generator is enabled with `-G Ninja`, so `CMAKE_MAKE_PROGRAM` is already set to `ninja`. Why is it required?

Author: This pr is about deps for arch and fedora only please keep on topic thanks

Reviewer 3: Yes, it is, but you've added an unneeded define that wasn't there before, so I was pointing it out.
